File: php/features.php
<?php
declare(strict_types=1);

// TeeheeJoke SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TeeheeJokeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TeeheeJokeBaseFeature();
            case "test":
                return new TeeheeJokeTestFeature();
            default:
                return new TeeheeJokeBaseFeature();
        }
    }
}

File: py/features.py
# TeeheeJoke SDK feature factory

from feature.base_feature import TeeheeJokeBaseFeature
from feature.test_feature import TeeheeJokeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TeeheeJokeBaseFeature(),
        "test": lambda: TeeheeJokeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
